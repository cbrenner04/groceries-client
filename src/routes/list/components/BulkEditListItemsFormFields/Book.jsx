import React from 'react';
import PropTypes from 'prop-types';

import { CheckboxField, TextField } from '../../../../components/FormFields';

const Book = ({ author, clearAuthor, handleClearAuthor, handleInput }) => (
  <TextField
    name="author"
    label="Author"
    value={author}
    handleChange={handleInput}
    placeholder="Kurt Vonnagut"
    disabled={clearAuthor}
    child={
      <CheckboxField
        name="clearAuthor"
        label="Clear author"
        handleChange={handleClearAuthor}
        value={clearAuthor}
        classes="ml-1 mt-1"
      />
    }
  />
);

Book.propTypes = {
  author: PropTypes.string.isRequired,
  clearAuthor: PropTypes.bool.isRequired,
  handleClearAuthor: PropTypes.func.isRequired,
  handleInput: PropTypes.func.isRequired,
};

export default Book;
