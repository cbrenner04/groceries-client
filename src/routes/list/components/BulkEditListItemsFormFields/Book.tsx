import React, { type ChangeEventHandler } from 'react';

import { CheckboxField, TextField } from 'components/FormFields';

export interface IBookProps {
  author: string;
  clearAuthor: boolean;
  handleClearAuthor: ChangeEventHandler;
  handleInput: ChangeEventHandler;
}

const Book: React.FC<IBookProps> = (props): React.JSX.Element => (
  <TextField
    name="author"
    label="Author"
    value={props.author}
    handleChange={props.handleInput}
    placeholder="Kurt Vonnagut"
    disabled={props.clearAuthor}
    child={
      <CheckboxField
        name="clearAuthor"
        label="Clear author"
        handleChange={props.handleClearAuthor}
        value={props.clearAuthor}
        classes="ms-1 mt-1"
      />
    }
  />
);

export default Book;
