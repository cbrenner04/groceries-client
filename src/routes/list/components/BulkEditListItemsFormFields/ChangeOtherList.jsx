import React from 'react';
import PropTypes from 'prop-types';
import { Button, Form } from 'react-bootstrap';

import { CheckboxField, SelectField, TextField } from '../../../../components/FormFields';
import { prettyListType } from '../../../../utils/format';

const ChangeOtherList = (props) => {
  const handleMove = () => {
    props.handleOtherListChange(false);
  };

  const handleCopy = () => {
    props.handleOtherListChange(true);
  };

  const changeListInstructions = props.existingListsOptions.length
    ? 'Choose an existing list or create a new one.'
    : `You do not have any other ${prettyListType(props.listType)}s. Please create a new list to take this action.`;

  return (
    <>
      <Form.Group className="form-check-inline mb-3">
        <Form.Check
          name="copy"
          label="Copy"
          type="radio"
          onChange={handleCopy}
          onClick={handleCopy}
          value={props.copy}
          id="move-action-copy"
          checked={props.copy}
        />
        <Form.Check
          name="move"
          label="Move"
          type="radio"
          value={props.move}
          id="move-action-move"
          onChange={handleMove}
          onClick={handleMove}
          checked={props.move}
        />
      </Form.Group>
      {(props.copy || props.move) && (
        <>
          <div className="mb-3">{changeListInstructions}</div>
          {props.showNewListForm && props.existingListsOptions.length > 0 && (
            <Button variant="link" onClick={props.clearNewListForm} className="mb-3">
              Choose existing list
            </Button>
          )}
          {!props.showNewListForm && (
            <>
              <Button variant="link" onClick={props.handleShowNewListForm} className="mb-3">
                Create new list
              </Button>
              <SelectField
                name="existingList"
                label="Existing list"
                value={props.existingList}
                options={props.existingListsOptions}
                handleChange={props.handleInput}
                blankOption
              />
            </>
          )}
          {props.showNewListForm && (
            <TextField
              name="newListName"
              label="New list name"
              value={props.newListName}
              handleChange={props.handleInput}
              placeholder="My super cool list"
            />
          )}
          {props.copy && !props.allComplete && (
            <CheckboxField
              name="updateCurrentItems"
              label="Would you like to also update the current items?"
              handleChange={props.handleInput}
              value={props.updateCurrentItems}
              classes="mt-3"
            />
          )}
        </>
      )}
    </>
  );
};

ChangeOtherList.propTypes = {
  handleOtherListChange: PropTypes.func.isRequired,
  copy: PropTypes.bool.isRequired,
  move: PropTypes.bool.isRequired,
  showNewListForm: PropTypes.bool.isRequired,
  existingListsOptions: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
    }),
  ).isRequired,
  listType: PropTypes.string.isRequired,
  handleInput: PropTypes.func.isRequired,
  handleShowNewListForm: PropTypes.func.isRequired,
  clearNewListForm: PropTypes.func.isRequired,
  existingList: PropTypes.string.isRequired,
  newListName: PropTypes.string.isRequired,
  updateCurrentItems: PropTypes.bool.isRequired,
  allComplete: PropTypes.bool.isRequired,
};

export default ChangeOtherList;
