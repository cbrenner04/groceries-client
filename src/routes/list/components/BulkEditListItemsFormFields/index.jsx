import React from 'react';
import PropTypes from 'prop-types';

import Book from './Book';
import Grocery from './Grocery';
import Music from './Music';
import ToDo from './ToDo';
import ChangeOtherList from './ChangeOtherList';
import { CategoryField, CheckboxField } from '../../../../components/FormFields';
import { listUsers } from '../../../../types';

const BulkEditListItemsFormFields = (props) => {
  return (
    <>
      <div>Move or copy these items to another list.</div>
      <br />
      <ChangeOtherList
        handleOtherListChange={props.handleOtherListChange}
        copy={props.formData.copy}
        move={props.formData.move}
        showNewListForm={props.formData.showNewListForm}
        existingListsOptions={props.existingListsOptions}
        listType={props.listType}
        handleInput={props.handleInput}
        handleShowNewListForm={props.handleShowNewListForm}
        clearNewListForm={props.clearNewListForm}
        existingList={props.formData.existingList}
        newListName={props.formData.newListName}
        updateCurrentItems={props.formData.updateCurrentItems}
        allComplete={props.formData.allComplete}
      />
      <hr />
      {!props.formData.allComplete && (
        <>
          <div>Update attributes for all items.</div>
          <br />
          {
            {
              BookList: (
                <Book
                  author={props.formData.author}
                  clearAuthor={props.formData.clearAuthor}
                  handleClearAuthor={() => props.clearAttribute('author', 'clearAuthor')}
                  handleInput={props.handleInput}
                />
              ),
              GroceryList: (
                <Grocery
                  quantity={props.formData.quantity}
                  clearQuantity={props.formData.clearQuantity}
                  handleClearQuantity={() => props.clearAttribute('quantity', 'clearQuantity')}
                  handleInput={props.handleInput}
                />
              ),
              MusicList: (
                <Music
                  artist={props.formData.artist}
                  clearArtist={props.formData.clearArtist}
                  handleClearArtist={() => props.clearAttribute('artist', 'clearArtist')}
                  album={props.formData.album}
                  clearAlbum={props.formData.clearAlbum}
                  handleClearAlbum={() => props.clearAttribute('album', 'clearAlbum')}
                  handleInput={props.handleInput}
                />
              ),
              // simple list has no updatable attributes
              ToDoList: (
                <ToDo
                  dueBy={props.formData.dueBy}
                  clearDueBy={props.formData.clearDueBy}
                  handleClearDueBy={() => props.clearAttribute('dueBy', 'clearDueBy')}
                  assigneeId={props.formData.assigneeId}
                  clearAssignee={props.formData.clearAssignee}
                  handleClearAssignee={() => props.clearAttribute('assigneeId', 'clearAssignee')}
                  handleInput={props.handleInput}
                  listUsers={props.listUsers}
                />
              ),
            }[props.listType]
          }
          <CategoryField
            name="category"
            category={props.formData.category}
            categories={props.categories}
            handleInput={props.handleInput}
            disabled={props.formData.clearCategory}
            child={
              <CheckboxField
                name="clearCategory"
                label="Clear category"
                handleChange={() => props.clearAttribute('category', 'clearCategory')}
                value={props.formData.clearCategory}
                classes="ms-1 mt-1"
              />
            }
          />
        </>
      )}
    </>
  );
};

BulkEditListItemsFormFields.propTypes = {
  listType: PropTypes.string.isRequired,
  formData: PropTypes.shape({
    copy: PropTypes.bool.isRequired,
    move: PropTypes.bool.isRequired,
    existingList: PropTypes.string,
    newListName: PropTypes.string,
    updateCurrentItems: PropTypes.bool.isRequired,
    album: PropTypes.string,
    clearAlbum: PropTypes.bool.isRequired,
    artist: PropTypes.string,
    clearArtist: PropTypes.bool.isRequired,
    assigneeId: PropTypes.string,
    clearAssignee: PropTypes.bool.isRequired,
    author: PropTypes.string,
    clearAuthor: PropTypes.bool.isRequired,
    category: PropTypes.string,
    clearCategory: PropTypes.bool.isRequired,
    dueBy: PropTypes.string,
    clearDueBy: PropTypes.bool.isRequired,
    quantity: PropTypes.string,
    clearQuantity: PropTypes.bool.isRequired,
    showNewListForm: PropTypes.bool.isRequired,
    allComplete: PropTypes.bool.isRequired,
  }).isRequired,
  handleInput: PropTypes.func.isRequired,
  clearAttribute: PropTypes.func.isRequired,
  listUsers: PropTypes.arrayOf(listUsers).isRequired,
  handleOtherListChange: PropTypes.func.isRequired,
  existingListsOptions: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
    }),
  ).isRequired,
  handleShowNewListForm: PropTypes.func.isRequired,
  clearNewListForm: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default BulkEditListItemsFormFields;
