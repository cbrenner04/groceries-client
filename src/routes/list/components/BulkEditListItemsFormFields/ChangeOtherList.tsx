import React, { type ChangeEventHandler } from 'react';
import { Button, Form } from 'react-bootstrap';

import { CheckboxField, SelectField, TextField } from 'components/FormFields';
import { prettyListType } from 'utils/format';

export interface IChangeOtherListProps {
  handleOtherListChange: (isCopy: boolean) => void;
  copy: boolean;
  move: boolean;
  showNewListForm: boolean;
  existingListsOptions: {
    value: string;
    label: string;
  }[];
  listType: string;
  handleInput: ChangeEventHandler;
  handleShowNewListForm: () => void;
  clearNewListForm: () => void;
  existingList: string;
  newListName: string;
  updateCurrentItems: boolean;
  allComplete: boolean;
}

const ChangeOtherList: React.FC<IChangeOtherListProps> = (props): React.JSX.Element => {
  const handleMove = (): void => {
    props.handleOtherListChange(false);
  };

  const handleCopy = (): void => {
    props.handleOtherListChange(true);
  };

  const changeListInstructions = props.existingListsOptions.length
    ? 'Choose an existing list or create a new one.'
    : `You do not have any other ${prettyListType(props.listType)}s. Please create a new list to take this action.`;

  return (
    <React.Fragment>
      <Form.Group className="form-check-inline mb-3">
        <Form.Check
          name="copy"
          label="Copy"
          type="radio"
          onChange={handleCopy}
          onClick={handleCopy}
          value={String(props.copy)}
          id="move-action-copy"
          checked={props.copy}
        />
        <Form.Check
          name="move"
          label="Move"
          type="radio"
          value={String(props.move)}
          id="move-action-move"
          onChange={handleMove}
          onClick={handleMove}
          checked={props.move}
        />
      </Form.Group>
      {(props.copy || props.move) && (
        <React.Fragment>
          <div className="mb-3">{changeListInstructions}</div>
          {props.showNewListForm && props.existingListsOptions.length > 0 && (
            <Button variant="link" onClick={props.clearNewListForm} className="mb-3">
              Choose existing list
            </Button>
          )}
          {!props.showNewListForm && (
            <React.Fragment>
              <Button variant="link" onClick={props.handleShowNewListForm} className="mb-3">
                Create new list
              </Button>
              <SelectField
                name="existingList"
                label="Existing list"
                value={props.existingList}
                options={props.existingListsOptions}
                handleChange={props.handleInput}
                blankOption
              />
            </React.Fragment>
          )}
          {props.showNewListForm && (
            <TextField
              name="newListName"
              label="New list name"
              value={props.newListName}
              handleChange={props.handleInput}
              placeholder="My super cool list"
            />
          )}
          {props.copy && !props.allComplete && (
            <CheckboxField
              name="updateCurrentItems"
              label="Would you like to also update the current items?"
              handleChange={props.handleInput}
              value={props.updateCurrentItems}
              classes="mt-3"
            />
          )}
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

export default ChangeOtherList;
