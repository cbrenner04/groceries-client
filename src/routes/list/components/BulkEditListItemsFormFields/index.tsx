import React, { type ChangeEventHandler } from 'react';

import { CategoryField, CheckboxField } from 'components/FormFields';
import { type IListUser } from 'typings';

import Book from './Book';
import Grocery from './Grocery';
import Music from './Music';
import ToDo from './ToDo';
import ChangeOtherList from './ChangeOtherList';

export interface IBulkEditListItemsFormFieldsFormDataProps {
  copy: boolean;
  move: boolean;
  existingList: string;
  newListName: string;
  updateCurrentItems: boolean;
  album?: string;
  clearAlbum: boolean;
  artist?: string;
  clearArtist: boolean;
  assigneeId?: string;
  clearAssignee: boolean;
  author?: string;
  clearAuthor: boolean;
  category?: string;
  clearCategory: boolean;
  dueBy?: string;
  clearDueBy: boolean;
  quantity?: string;
  clearQuantity: boolean;
  showNewListForm: boolean;
  allComplete: boolean;
}

export interface IBulkEditListItemsFormFieldsProps {
  listType: string;
  formData: IBulkEditListItemsFormFieldsFormDataProps;
  handleInput: ChangeEventHandler;
  clearAttribute: (
    attribute: keyof IBulkEditListItemsFormFieldsFormDataProps,
    clearAttribute: keyof IBulkEditListItemsFormFieldsFormDataProps,
  ) => void;
  listUsers: IListUser[];
  handleOtherListChange: (isCopy: boolean) => void;
  existingListsOptions: {
    value: string;
    label: string;
  }[];
  handleShowNewListForm: () => void;
  clearNewListForm: () => void;
  categories: string[];
}

const BulkEditListItemsFormFields: React.FC<IBulkEditListItemsFormFieldsProps> = (props): React.JSX.Element => {
  return (
    <React.Fragment>
      <div>Move or copy these items to another list.</div>
      <br />
      <ChangeOtherList
        handleOtherListChange={props.handleOtherListChange}
        copy={props.formData.copy}
        move={props.formData.move}
        showNewListForm={props.formData.showNewListForm}
        existingListsOptions={props.existingListsOptions}
        listType={props.listType}
        handleInput={props.handleInput}
        handleShowNewListForm={props.handleShowNewListForm}
        clearNewListForm={props.clearNewListForm}
        existingList={props.formData.existingList}
        newListName={props.formData.newListName}
        updateCurrentItems={props.formData.updateCurrentItems}
        allComplete={props.formData.allComplete}
      />
      <hr />
      {!props.formData.allComplete && (
        <React.Fragment>
          <div>Update attributes for all items.</div>
          <br />
          {
            {
              BookList: (
                <Book
                  author={props.formData.author ?? ''}
                  clearAuthor={props.formData.clearAuthor}
                  handleClearAuthor={(): void => props.clearAttribute('author', 'clearAuthor')}
                  handleInput={props.handleInput}
                />
              ),
              GroceryList: (
                <Grocery
                  quantity={props.formData.quantity ?? ''}
                  clearQuantity={props.formData.clearQuantity}
                  handleClearQuantity={(): void => props.clearAttribute('quantity', 'clearQuantity')}
                  handleInput={props.handleInput}
                />
              ),
              MusicList: (
                <Music
                  artist={props.formData.artist ?? ''}
                  clearArtist={props.formData.clearArtist}
                  handleClearArtist={(): void => props.clearAttribute('artist', 'clearArtist')}
                  album={props.formData.album ?? ''}
                  clearAlbum={props.formData.clearAlbum}
                  handleClearAlbum={(): void => props.clearAttribute('album', 'clearAlbum')}
                  handleInput={props.handleInput}
                />
              ),
              // simple list has no updatable attributes
              ToDoList: (
                <ToDo
                  dueBy={props.formData.dueBy ?? ''}
                  clearDueBy={props.formData.clearDueBy}
                  handleClearDueBy={(): void => props.clearAttribute('dueBy', 'clearDueBy')}
                  assigneeId={props.formData.assigneeId ?? ''}
                  clearAssignee={props.formData.clearAssignee}
                  handleClearAssignee={(): void => props.clearAttribute('assigneeId', 'clearAssignee')}
                  handleInput={props.handleInput}
                  listUsers={props.listUsers}
                />
              ),
            }[props.listType]
          }
          <CategoryField
            name="category"
            category={props.formData.category}
            categories={props.categories}
            handleInput={props.handleInput}
            disabled={props.formData.clearCategory}
            child={
              <CheckboxField
                name="clearCategory"
                label="Clear category"
                handleChange={(): void => props.clearAttribute('category', 'clearCategory')}
                value={props.formData.clearCategory}
                classes="ms-1 mt-1"
              />
            }
          />
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

export default BulkEditListItemsFormFields;
