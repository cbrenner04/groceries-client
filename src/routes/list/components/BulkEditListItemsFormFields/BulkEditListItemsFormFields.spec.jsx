import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import BulkEditListItemsFormFields from './index';

describe('BulkEditListItemsFormFields', () => {
  const props = {
    listType: 'GroceryList',
    formData: {
      copy: false,
      move: false,
      existingList: '',
      newListName: '',
      updateCurrentItems: false,
      album: '',
      clearAlbum: false,
      artist: '',
      clearArtist: false,
      assigneeId: '',
      clearAssignee: false,
      author: '',
      clearAuthor: false,
      category: '',
      clearCategory: false,
      dueBy: '',
      clearDueBy: false,
      quantity: '',
      clearQuantity: false,
      showNewListForm: false,
      allComplete: false,
    },
    handleInput: jest.fn(),
    clearAttribute: jest.fn(),
    listUsers: [
      {
        id: 1,
        email: '',
      },
    ],
    handleOtherListChange: jest.fn(),
    existingListsOptions: [
      {
        value: '1',
        label: 'Foo',
      },
    ],
    handleShowNewListForm: jest.fn(),
    clearNewListForm: jest.fn(),
    categories: ['foo'],
  };

  it('render Book fields when listType is BookList and fires appropriate change handlers', () => {
    props.listType = 'BookList';
    const { container, getByLabelText, getAllByRole } = render(<BulkEditListItemsFormFields {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Author')).toBeTruthy();

    fireEvent.change(getByLabelText('Author'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Category'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[0]);

    expect(props.clearAttribute).toHaveBeenCalledWith('author', 'clearAuthor');

    fireEvent.click(getAllByRole('checkbox')[1]);

    expect(props.clearAttribute).toHaveBeenCalledWith('category', 'clearCategory');
  });

  it('render Grocery fields when listType is GroceryList and fires appropriate change handlers', () => {
    props.listType = 'GroceryList';
    const { container, getByLabelText, getAllByRole } = render(<BulkEditListItemsFormFields {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Quantity')).toBeTruthy();

    fireEvent.change(getByLabelText('Quantity'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Category'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[0]);

    expect(props.clearAttribute).toHaveBeenCalledWith('quantity', 'clearQuantity');

    fireEvent.click(getAllByRole('checkbox')[1]);

    expect(props.clearAttribute).toHaveBeenCalledWith('category', 'clearCategory');
  });

  it('render Music fields when listType is MusicList and fires appropriate change handlers', () => {
    props.listType = 'MusicList';
    const { container, getByLabelText, getAllByRole } = render(<BulkEditListItemsFormFields {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Album')).toBeTruthy();

    fireEvent.change(getByLabelText('Album'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Category'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[0]);

    expect(props.clearAttribute).toHaveBeenCalledWith('artist', 'clearArtist');

    fireEvent.click(getAllByRole('checkbox')[1]);

    expect(props.clearAttribute).toHaveBeenCalledWith('album', 'clearAlbum');

    fireEvent.click(getAllByRole('checkbox')[2]);

    expect(props.clearAttribute).toHaveBeenCalledWith('category', 'clearCategory');
  });

  it('render ToDo fields when listType is ToDoList and fires appropriate change handlers', () => {
    props.listType = 'ToDoList';
    const { container, getByLabelText, getAllByRole } = render(<BulkEditListItemsFormFields {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByLabelText('Assignee')).toBeTruthy();

    fireEvent.change(getByLabelText('Assignee'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Category'), { target: { value: 'a' } });

    expect(props.handleInput).toHaveBeenCalled();

    fireEvent.click(getAllByRole('checkbox')[0]);

    expect(props.clearAttribute).toHaveBeenCalledWith('assigneeId', 'clearAssignee');

    fireEvent.click(getAllByRole('checkbox')[1]);

    expect(props.clearAttribute).toHaveBeenCalledWith('dueBy', 'clearDueBy');

    fireEvent.click(getAllByRole('checkbox')[2]);

    expect(props.clearAttribute).toHaveBeenCalledWith('category', 'clearCategory');
  });

  it('does not render list item attribute fields when all items are completed', () => {
    props.listType = 'GroceryList';
    props.formData.allComplete = true;
    const { container, queryByLabelText } = render(<BulkEditListItemsFormFields {...props} />);

    expect(container).toMatchSnapshot();
    expect(queryByLabelText('Quantity')).toBeNull();
  });
});
