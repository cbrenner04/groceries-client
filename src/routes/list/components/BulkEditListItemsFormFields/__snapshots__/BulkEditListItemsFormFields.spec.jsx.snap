// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkEditListItemsFormFields does not render list item attribute fields when all items are completed 1`] = `
<div>
  <div>
    Move or copy these items to another list.
  </div>
  <br />
  <div
    class="form-check-inline mb-3"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-copy"
        name="copy"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-copy"
        title=""
      >
        Copy
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-move"
        name="move"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-move"
        title=""
      >
        Move
      </label>
    </div>
  </div>
  <hr />
</div>
`;

exports[`BulkEditListItemsFormFields render Book fields when listType is BookList and fires appropriate change handlers 1`] = `
<div>
  <div>
    Move or copy these items to another list.
  </div>
  <br />
  <div
    class="form-check-inline mb-3"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-copy"
        name="copy"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-copy"
        title=""
      >
        Copy
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-move"
        name="move"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-move"
        title=""
      >
        Move
      </label>
    </div>
  </div>
  <hr />
  <div>
    Update attributes for all items.
  </div>
  <br />
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="author"
    >
      Author
    </label>
    <input
      class="form-control"
      id="author"
      name="author"
      placeholder="Kurt Vonnagut"
      type="text"
      value=""
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAuthor"
          name="clearAuthor"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAuthor"
          title=""
        >
          Clear author
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="category"
    >
      Category
    </label>
    <input
      class="form-control"
      id="category"
      list="categories"
      name="category"
      type="text"
      value=""
    />
    <datalist
      id="categories"
    >
      <option
        value="foo"
      />
    </datalist>
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearCategory"
          name="clearCategory"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearCategory"
          title=""
        >
          Clear category
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`BulkEditListItemsFormFields render Grocery fields when listType is GroceryList and fires appropriate change handlers 1`] = `
<div>
  <div>
    Move or copy these items to another list.
  </div>
  <br />
  <div
    class="form-check-inline mb-3"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-copy"
        name="copy"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-copy"
        title=""
      >
        Copy
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-move"
        name="move"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-move"
        title=""
      >
        Move
      </label>
    </div>
  </div>
  <hr />
  <div>
    Update attributes for all items.
  </div>
  <br />
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="quantity"
    >
      Quantity
    </label>
    <input
      class="form-control"
      id="quantity"
      name="quantity"
      placeholder="3 bags"
      type="text"
      value=""
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearQuantity"
          name="clearQuantity"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearQuantity"
          title=""
        >
          Clear quantity
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="category"
    >
      Category
    </label>
    <input
      class="form-control"
      id="category"
      list="categories"
      name="category"
      type="text"
      value=""
    />
    <datalist
      id="categories"
    >
      <option
        value="foo"
      />
    </datalist>
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearCategory"
          name="clearCategory"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearCategory"
          title=""
        >
          Clear category
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`BulkEditListItemsFormFields render Music fields when listType is MusicList and fires appropriate change handlers 1`] = `
<div>
  <div>
    Move or copy these items to another list.
  </div>
  <br />
  <div
    class="form-check-inline mb-3"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-copy"
        name="copy"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-copy"
        title=""
      >
        Copy
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-move"
        name="move"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-move"
        title=""
      >
        Move
      </label>
    </div>
  </div>
  <hr />
  <div>
    Update attributes for all items.
  </div>
  <br />
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="artist"
    >
      Artist
    </label>
    <input
      class="form-control"
      id="artist"
      name="artist"
      placeholder="Sir Mix-a-Lot"
      type="text"
      value=""
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearArtist"
          name="clearArtist"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearArtist"
          title=""
        >
          Clear artist
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="album"
    >
      Album
    </label>
    <input
      class="form-control"
      id="album"
      name="album"
      placeholder="Mack Daddy"
      type="text"
      value=""
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAlbum"
          name="clearAlbum"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAlbum"
          title=""
        >
          Clear album
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="category"
    >
      Category
    </label>
    <input
      class="form-control"
      id="category"
      list="categories"
      name="category"
      type="text"
      value=""
    />
    <datalist
      id="categories"
    >
      <option
        value="foo"
      />
    </datalist>
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearCategory"
          name="clearCategory"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearCategory"
          title=""
        >
          Clear category
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`BulkEditListItemsFormFields render ToDo fields when listType is ToDoList and fires appropriate change handlers 1`] = `
<div>
  <div>
    Move or copy these items to another list.
  </div>
  <br />
  <div
    class="form-check-inline mb-3"
  >
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-copy"
        name="copy"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-copy"
        title=""
      >
        Copy
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input"
        id="move-action-move"
        name="move"
        type="radio"
        value="false"
      />
      <label
        class="form-check-label"
        for="move-action-move"
        title=""
      >
        Move
      </label>
    </div>
  </div>
  <hr />
  <div>
    Update attributes for all items.
  </div>
  <br />
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="assigneeId"
    >
      Assignee
    </label>
    <select
      class="form-control"
      id="assigneeId"
      name="assigneeId"
    >
      <option
        disabled=""
        value=""
      >
        Select Assignee
      </option>
      <option
        value="id1"
      />
    </select>
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearAssignee"
          name="clearAssignee"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearAssignee"
          title=""
        >
          Clear assignee
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="dueBy"
    >
      Due By
    </label>
    <input
      class="form-control"
      id="dueBy"
      name="dueBy"
      placeholder="mm/dd/yyyy"
      type="date"
      value=""
    />
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearDueBy"
          name="clearDueBy"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearDueBy"
          title=""
        >
          Clear due by
        </label>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
      for="category"
    >
      Category
    </label>
    <input
      class="form-control"
      id="category"
      list="categories"
      name="category"
      type="text"
      value=""
    />
    <datalist
      id="categories"
    >
      <option
        value="foo"
      />
    </datalist>
    <div
      class="ms-1 mt-1 mb-3"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="clearCategory"
          name="clearCategory"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="clearCategory"
          title=""
        >
          Clear category
        </label>
      </div>
    </div>
  </div>
</div>
`;
